tests/test_raster.py

```diff
@@ -1381,24 +1381,18 @@ def test_reproject(self, example: str) -> None:
             assert (out_img.count, *out_img.shape) == (n, 500, 500)
 
         # Test that the rounding of resolution is correct for large rasters
-        # (we take an example that used to fail, see issue #354)
-        data = np.zeros(shape=(5741, 2959), dtype="uint8")
-        transform = rio.transform.Affine(20.0, 0.0, 238286.29553975424, 0.0, -20.0, 6995453.456051373)
+        # (we take an example that used to fail, see issue #354 and #357)
+        data = np.ones((4759, 2453))
+        transform = rio.transform.Affine(
+            24.12423878332849, 0.0, 238286.29553975424, 0.0, -24.12423878332849, 6995453.456051373
+        )
         crs = rio.CRS.from_epsg(32633)
         nodata = -9999.0
         rst = gu.Raster.from_array(data=data, transform=transform, crs=crs, nodata=nodata)
 
-        # This large grid geotransform is taken from https://cdn.proj.org/us_nga_egm08_25.tif
```

Comment:
Why not leave this previous example as well?

Reply:
In your example, you created two rasters (`rst` and `rst2`), one to be reprojected on the same grid as the other, to test the option with match-reference. I find these examples quite specific because the transforms are provided. I found it easier to create only one and then reproject to given bounds and res. If this one case works, by definition the match-reference case works because now this is exactly what is used for the match-reference functionality. Is that ok?

Reply:
OK makes sense!